<?xml version="1.0"?>
<launch>
    <arg name="config_path" default="$(find multi_localizer)/config" />
	<arg name="roomba"    default="roomba1" />

	<!-- rosbag -->
	<arg name="date"      default="20220224" />
	<!-- <arg name="date"      default="20220630" /> -->
	<arg name="is_bag" default="false" />
	<group if="$(arg is_bag)" >
		<include file="$(find multi_localizer)/launch/utils/bag_player.launch" >
			<arg name="roomba" value="$(arg roomba)" />
			<arg name="date"   value="$(arg date)" />
		</include>
	</group>

	<!--map_server -->
	<arg name="is_map" default="false" />
	<group if="$(arg is_map)" >
		<node pkg="map_server" type="map_server" name="map" args="$(arg config_path)/map/spare/map.yaml" />
	</group>

	<!-- tf -->
	<include file="$(find multi_localizer)/launch/tf_broadcaster.launch" >
        <arg name="roomba" value="$(arg roomba)"/>
    </include>

	<!-- amcl_pose_republisher -->
	<arg name="publish_obj_msg" default="true" />
	<node pkg="multi_localizer" type="amcl_pose_republisher" name="$(arg roomba)_amcl_pose_republisher" >
		<!-- remap -->
		<remap from="pose_in"  to="$(arg roomba)/amcl_pose" />
		<remap from="odom_in"  to="$(arg roomba)/roomba/odometry" />
		<remap from="obj_in"   to="$(arg roomba)/object_positions" />
		<remap from="pose_out" to="$(arg roomba)/pose" />
		<remap from="obj_out"  to="$(arg roomba)/objects_data" />
		
		<!-- params -->
		<param name="MAP_FRAME_ID"       value="map" />
		<param name="BASE_LINK_FRAME_ID" value="$(arg roomba)/base_link" />
		<param name="PUBLISH_OBJ_MSG"    value="$(arg publish_obj_msg)" />
	</node>

	<!-- rviz -->
	<arg name="is_rviz" default="false" />
    <group if="$(arg is_rviz)" >
        <node pkg="rviz" type="rviz" name="$(arg roomba)_rviz" args="-d $(arg config_path)/rviz/amcl_pose_republisher/$(arg roomba).rviz" />
    </group>

</launch>